archive_upload/lib/jobrunner.py

```diff
@@ -0,0 +1,150 @@
+
+    def stop(self, job_id):
+        """
+        Stop job with job_id
+        :param job_id: of job to stop
+        :return: the job_id of the stopped job, or None if not found.
+        """
+        raise NotImplementedError("Subclasses should implement this!")
+
+    def stop_all(self):
+        """
+        Stop all jobs
+        :return: Nothing
+        """
+        raise NotImplementedError("Subclasses should implement this!")
+
+    def status(self, job_id):
+        """
+        Status of job with id
+        :param job_id: to get status for.
+        :return: It's status
+        """
+        raise NotImplementedError("Subclasses should implement this!")
+
+    def status_all(self):
+        """
+        Get status for all jobs
+        :param job_id: to get status for.
+        :return: A dict containing all jobs with job_id as key and status as value.
+        """
+        raise NotImplementedError("Subclasses should implement this!")
+
+
+class LocalQAdapter(JobRunnerAdapter):
+    """
+    An implementation of `JobRunnerAdapter` running jobs through
+    localq (a jobrunner which will schedule jobs on a single node).
+    """
+
+    @staticmethod
+    def localq2arteria_status(status):
+        """
+        Convert a localq status to an arteria state
+        :param status: to convert
+        :return: the arteria state
+        """
+
+        if status == Status.COMPLETED:
+            return arteria_state.DONE
+        elif status == Status.FAILED:
+            return arteria_state.ERROR
+        elif status == Status.PENDING:
+            return arteria_state.PENDING
+        elif status == Status.RUNNING:
+            return arteria_state.STARTED
+        elif status == Status.CANCELLED:
+            return arteria_state.CANCELLED
+        elif status == Status.NOT_FOUND:
+            return arteria_state.NONE
+        else:
+            return arteria_state.NONE
+
+    def __init__(self, nbr_of_cores, whitelisted_warnings, interval=30, priority_method="fifo"):
+        self.nbr_of_cores = nbr_of_cores
+        self.whitelisted_warnings = whitelisted_warnings
+        self.server = LocalQServer(nbr_of_cores, interval, priority_method, use_shell=True)
+        self.server.run()
+
+    def start(self, cmd, nbr_of_cores, run_dir, stdout=None, stderr=None):
+        return self.server.add(cmd, nbr_of_cores, run_dir, stdout=stdout, stderr=stderr)
+
+    def stop(self, job_id):
+        return self.server.stop_job_with_id(job_id)
+
+    def stop_all(self):
+        return self.server.stop_all_jobs()
+
+    # Returns the stats of the long running DSMC or md5sum job. 
+    def status(self, job_id):
+        arteria_status = LocalQAdapter.localq2arteria_status(self.server.get_status(job_id))
+
+        # This is a bit hacky, because we're assuming this will only/mostly happen to 
+        # dsmc and not md5sum. 
+        if arteria_status == arteria_state.ERROR: 
```

Comment:
I'm not sure, but perhaps you could access the job here and check which command is used? And in that way check if it is a dsmc job or not?

Reply:
I'll improve it a little bit.